lib/factor/factorQuadratic.js

```diff
@@ -153,61 +144,57 @@ function factorDifferenceOfSquares(node, symbol, aValue, bValue, cValue, negate)
 // are perfect squares and b = 2*sqrt(a)*sqrt(c)
 // e.g. x^2 + 2x + 1 -> (x + 1)^2
 function factorPerfectSquare(node, symbol, aValue, bValue, cValue, negate) {
-  // check if perfect square: are a and c squares and b = 2*sqrt(a)*sqrt(c)
-  if (bValue && cValue) {
-    const aRootValue = Math.sqrt(Math.abs(aValue));
-    let cRootValue = Math.sqrt(Math.abs(cValue));
-
-    // if the second term is negative, then the constant in the parens is
-    // subtracted: e.g. x^2 - 2x + 1 -> (x - 1)^2
-    if (bValue < 0) {
-      cRootValue = cRootValue * -1;
-    }
+  // check if perfect square: (i) a and c squares, (ii) b = 2*sqrt(a)*sqrt(c)
+  if (!bValue || !cValue) {
+    return Node.Status.noChange(node);
+  }
 
-    // apply the perfect square test
-    const perfectProduct = 2 * aRootValue * cRootValue;
-    if (Number.isInteger(aRootValue) &&
-        Number.isInteger(cRootValue) &&
-        bValue === perfectProduct) {
+  const aRootValue = Math.sqrt(Math.abs(aValue));
+  let cRootValue = Math.sqrt(Math.abs(cValue));
 
-      const aRootNode = Node.Creator.constant(aRootValue);
-      const cRootNode = Node.Creator.constant(cRootValue);
+  // if the second term is negative, then the constant in the parens is
+  // subtracted: e.g. x^2 - 2x + 1 -> (x - 1)^2
+  if (bValue < 0) {
+    cRootValue = cRootValue * -1;
+  }
 
-      const polyTerm = Node.Creator.polynomialTerm(
-        symbol, null, aRootNode);
-      let paren = Node.Creator.parenthesis(
-        Node.Creator.operator('+', [polyTerm, cRootNode]));
-      const exponent = Node.Creator.constant(2);
+  // apply the perfect square test
+  const perfectProduct = 2 * aRootValue * cRootValue;
+  if (Number.isInteger(aRootValue) &&
+      Number.isInteger(cRootValue) &&
+      bValue === perfectProduct) {
 
-      if (negate) {
-        paren = Negative.negate(paren);
-      }
+    const aRootNode = Node.Creator.constant(aRootValue);
+    const cRootNode = Node.Creator.constant(cRootValue);
 
-      // create node in perfect square form
-      const newNode = Node.Creator.operator('^', [paren, exponent]);
-      return Node.Status.nodeChanged(
-        ChangeTypes.FACTOR_PERFECT_SQUARE, node, newNode);
+    const polyTerm = Node.Creator.polynomialTerm(symbol, null, aRootNode);
+    let paren = Node.Creator.parenthesis(
+      Node.Creator.operator('+', [polyTerm, cRootNode]));
+    const exponent = Node.Creator.constant(2);
+
+    if (negate) {
+      paren = Negative.negate(paren);
```

Comment:
if you negate before doing the ^2, won't the negating cancel out?

ie `(-(x+2))^2`

I see your test passes though, which confuses me